src/Marten/Schema/Identity/Sequences/HiLoSequence.cs

```diff
@@ -91,14 +81,26 @@ public void AdvanceToNextHi()
 
                 try
                 {
-                    var tx = conn.BeginTransaction(IsolationLevel.Serializable);
-                    var raw = conn.CreateCommand().CallsSproc(GetNextFunction)
-                        .With("entity", _entityName)
-                        .Returns("next", NpgsqlDbType.Bigint).ExecuteScalar();
+                    var attempts = 0;
+                    do
+                    {
+                        attempts++;
+
+                        // Sproc is expected to return -1 if it's unable to
+                        // atomically secure the next hi
+                        var raw = conn.CreateCommand().CallsSproc(GetNextFunction)
+                            .With("entity", _entityName)
+                            .Returns("next", NpgsqlDbType.Bigint).ExecuteScalar();
+
+                        CurrentHi = Convert.ToInt64(raw);
 
-                    tx.Commit();
+                    } while (CurrentHi < 0 && attempts < 30);
 
-                    CurrentHi = Convert.ToInt64(raw);
+                    // if CurrentHi is still less than 0 at this point, then throw exception
+                    if (CurrentHi < 0)
+                    {
+                        throw new Exception("Unable to advance hi sequence");
```

Comment:
As you indicated, a specific exception for this case will be better rather than generic exception.

Reply:
Added exception `HiloSequenceAdvanceToNextHiAttemptsExceededException`